test/xpu_api/containers/sequences/array/array.comparison/greater_g.pass.cpp

```diff
@@ -42,12 +43,18 @@ kernel_test()
     }
     return ret;
 }
+#endif // !_PSTL_TEST_COMPARISON_BROKEN
 
 int
 main()
 {
+    bool tests_run = false;
+
+#if !_PSTL_TEST_COMPARISON_BROKEN
     auto ret = kernel_test();
     EXPECT_TRUE(ret, "Wrong result of work with dpl::array and '>' in kernel_test");
+    tests_run = true;
+#endif // !_PSTL_TEST_COMPARISON_BROKEN
 
-    return TestUtils::done();
+    return TestUtils::done(tests_run);
```

Comment:
I propose to change a little bit this code:
- option 1: rename `tests_run` to `bProcessed` like in other tests or wise versa;
- option 2: not introduce this local variable (here and in all other places and use `_PSTL_TEST_COMPARISON_BROKEN` here: return `TestUtils::done(_PSTL_TEST_COMPARISON_BROKEN);`
But redefine `_PSTL_TEST_COMPARISON_BROKEN` by another way:
```C++
// oneAPI DPC++ compiler 2024.2.0 and earlier is unable to eliminate a "dead" function call to an undefined function
// within a sycl kernel which MSVC uses to allow comparisons with literal zero without warning
#if __cplusplus >= 202002L && _MSVC_LANG >= 202002L && _MSVC_STL_VERSION >= 143 && _MSVC_STL_UPDATE >= 202303L && __INTEL_LLVM_COMPILER <= 20240200
#    define _PSTL_TEST_COMPARISON_BROKEN 1
#else
#    define _PSTL_TEST_COMPARISON_BROKEN 0
#endif
```

Reply:
Sure, I can do that.  I don't have a strong preference, I'll do option 1 if there are no objections.

Reply:
done (option 1)